app/controllers/webhooks_controller.rb

```diff
@@ -69,7 +69,11 @@ def email_params
   end
 
   def authorized?
-    params[:auth].present? && params[:auth].secure_compare(ENV['WEBHOOK_AUTH_KEY'])
+    params[:auth].present? &&
+      ActiveSupport::SecurityUtils.secure_compare(
+        params[:auth],
```

Comment:
According to [the documentation of `secure_compare`](http://api.rubyonrails.org/classes/ActiveSupport/SecurityUtils.html#method-c-secure_compare):

> This should not be used on variable length plaintext strings because it could leak length info via timing attacks.

That implies that we should do something to fix the length beforehand. I _think_ this would work:

```ruby
length = ENV.fetch('WEBHOOK_AUTH_KEY').length
fixed_length_param = (params.fetch(:auth) + ' ' * length).slice(0, length)
```

… but I don't know whether that has a fixed time cost.